Sources/NSDictionary+Safety.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+extension NSDictionary {
+    func safeValueForKeyPath(keyPath: String) -> AnyObject? {
+        var object: AnyObject? = self
+        var keys = keyPath.characters.split(".").map(String.init)
+
+        while keys.count > 0 {
```

Comment:
I'd add `let currentObject = object` here that way if you ask for `a.b.c.d.e.f.g.h.i` and b is already producing a `nil` result you don't continue